### PR TITLE
feat: Core Modularization & Sensor API

### PR DESCRIPTION
**Pull request by EffortlessSteven**
# PR Description: V0.2 Enhancements & Architecture Refactoring

## Overview
This PR introduces significant enhancements to the Diffguard ecosystem, including a major architectural refactoring, the introduction of a new Sensor API for integration with external dashboards (Cockpit), and various improvements to configuration management, reporting, and test coverage.

### 1. Architectural Refactoring: `diffguard-app` to `diffguard-core`
- **Migrated core logic:** Moved the primary analysis and reporting logic from the CLI-focused `diffguard-app` crate to a new, reusable `diffguard-core` crate.
- **Decoupling:** Decoupled the CLI implementation from the core engine to enable easier integration of Diffguard logic into other tools or services.
- **Improved Types:** Introduced `diffguard-types` to centralize domain models and ensure consistency across the workspace.

### 2. Sensor API & Cockpit Integration
- **New Sensor API:** Implemented a new reporting layer (`sensor.rs`, `sensor_api.rs`) that adheres to the `sensor.report.v1.schema.json` specification.
- **Cockpit Mode:** Added support for "cockpit mode" in the CLI, allowing Diffguard to generate reports specifically formatted for consumption by external monitoring dashboards.
- **Conformance Testing:** Added `xtask` commands and specialized tests (`conform_real.rs`) to ensure the generated sensor reports strictly follow the defined JSON schemas.

### 3. Configuration & Environment Enhancements
- **Environment Variable Expansion:** Configuration files now support environment variable expansion (e.g., `${MY_VAR}`), facilitating easier use in CI/CD pipelines.
- **Advanced Configuration Loader:** A new `config_loader.rs` provides more robust validation and merging of configuration files, including support for presets.
- **Schema Validation:** Introduced JSON schemas for `diffguard.config.json` and `diffguard.check.json` to provide better editor support and validation for users.

### 4. Reporting & Metrics
- **Enhanced Sensor Reports:** Reports now include detailed findings metrics, timing data, and structured verdict reasons.
- **Markdown & SARIF Improvements:** Refactored markdown rendering for better readability and ensured SARIF output remains compliant with industry standards.
- **JSON Receipts:** Added detailed JSON receipts for every check, including timing metrics and fingerprinting for deduplication.

### 5. Testing & Quality Assurance
- **Comprehensive Snapshot Testing:** Added a vast suite of snapshot tests (over 50 new snapshots) covering various reporting formats (JSON, Markdown, SARIF, GitHub Annotations) and edge cases.
- **BDD Integration Tests:** Introduced Cucumber-style BDD tests for CLI workflows to ensure end-to-end functionality.
- **Fingerprinting:** Updated finding fingerprinting to use full SHA-256 hashes for more reliable deduplication across runs.
- **CI/CD Integration:** Provided example workflow files for GitHub Actions and Azure Pipelines.

### 6. Repository Hygiene & Maintenance
- **Updated `.gitignore`:** Improved ignore patterns to exclude build artifacts (`target*`), PDBs, profiling data (`.profraw`), and temporary test directories.
- **History Cleanup:** Removed accidental binary and artifact commits from the local history during the rebase process.
- **Documentation:** Updated `ROADMAP.md`, `CHANGELOG.md`, and added `docs/cockpit-integration.md` to reflect the latest changes.

## Verification Results
- **Unit & Integration Tests:** All tests passing.
- **Snapshot Tests:** Verified and updated all snapshots to match the new architecture.
- **Schema Validation:** Sensor reports validated against `sensor.report.v1.schema.json`.
- **Linting:** Clippy warnings addressed.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,10 +14,10 @@ default-members = ["crates/diffguard"]
 
 [workspace.package]
 version = "0.1.0"
-authors = ["Effortless Metrics <contact@effortlessmetrics.com>"]
-edition = "2021"
+authors = ["Steven Zimmerman, CPA <git@effortlesssteven.com>"]
+edition = "2024"
 license = "MIT OR Apache-2.0"
-rust-version = "1.75"
+rust-version = "1.92"
```
[the remark this answers is not in the document]
Comment (EffortlessSteven) [reply Cargo.toml]: 2024 and 1.92 are stable.
